### DISCUSSION TITLE
Problem with non-idempotent manifests

### DISCUSSION THREAD

**Discussion by tschale** [Getdeck/getdeck#107] [Ideas]
I stumbled upon an issue with K8s manifests that don't seem to be idempotent. It is in principal the NFS, which can be found in https://github.com/Getdeck/wharf/tree/main/nfs However I can't seem to reproduce it with the one in wharf, only with that of the Ofa dev setup... If I run `get deck` twice, it somehow breaks the Statefulset and makes the volumes unavailable for the applications.

I just wanted to open the discussion here, as to whether we wan't to acknowledge that there might be idempotent manifests, or manifests that should only be run once. In theory we could handle that, if we label resources as deployed by deck and provide a flag in the Deckfile to indicate that a resource should only be applied once.
However I think it's totally up for discussion as that should even be a valid use-case, or if non-idempotent manifests just shouldn't exist. It could be interesting to see as to whether there might be some cluster setup manifests for which that could be relevant.

**Comment by buschNT** [comment]
I guess we could address this issue by adding some kind of "apply only after cluster creation" setting to the sources. Then, these sources should be applied only once. We would have to think about this a bit more in detail.

> However I think it's totally up for discussion as that should even be a valid use-case, or if non-idempotent manifests just shouldn't exist.

Even if non-idempotent manifests maybe shouldn't exist, we could still try to handle this issue. I guess sooner or later someone will run into the same issue. And maybe you are dependent on manifests you cannot control/adjust.

**Reply by tschale** [reply]
> Even if non-idempotent manifests maybe shouldn't exist, we could still try to handle this issue. I guess sooner or later someone will run into the same issue. And maybe you are dependent on manifests you cannot control/adjust.

That's a valid point. They are possible and so it's somewhat likely that at some point someone will use them.
